blocks/api/raw-handling/strip-attributes.js

```diff
@@ -20,10 +20,26 @@ export default function( node ) {
 	const tag = node.nodeName.toLowerCase();
 
 	Array.from( node.attributes ).forEach( ( { name } ) => {
-		if ( isAttributeWhitelisted( tag, name ) ) {
+		if ( name === 'class' || isAttributeWhitelisted( tag, name ) ) {
 			return;
 		}
 
 		node.removeAttribute( name );
 	} );
+
+	const oldClasses = node.getAttribute( 'class' );
+
+	if ( ! oldClasses ) {
+		return;
+	}
+
+	const newClasses = oldClasses.split( ' ' ).filter( ( name ) => {
+		return name && isClassWhitelisted( tag, name );
+	} );
```

Comment:
For type consistency with `oldClasses`, the join should happen here:

```js
const newClasses = oldClasses.split( ' ' ).filter( … ).join( ' ' );
```

also, that filter predicate is a great candidate for implicit `return`

```js
.filter( ( name ) => name && …
```

The subsequent `newClasses.length` would still work (switching from `Array#length` to `String#length`).